sklearn/datasets/_base.py

```diff
@@ -1501,14 +1504,37 @@ def _fetch_remote(remote, dirname=None):
     dirname : str
         Directory to save the file to.
 
+    n_retries : int, default=3
+        Number of retries when HTTP errors are encountered.
+
+        .. versionadded:: 1.2
+
+    delay : int, default=1
+        Number of seconds between retries.
+
+        .. versionadded:: 1.2
+
     Returns
     -------
     file_path: str
         Full path of the created file.
     """
 
     file_path = remote.filename if dirname is None else join(dirname, remote.filename)
-    urlretrieve(remote.url, file_path)
+
+    retry_cnt = n_retries
+    while 1:
+        try:
+            urlretrieve(remote.url, file_path)
+            break
+        except (URLError, TimeoutError):
+            if retry_cnt > 0:
+                warnings.warn(f"Retry downloading from url: {remote.url}")
+                time.sleep(delay)
+                retry_cnt -= 1
```

Comment:
Is there a particular reason to create and use `retry_cnt` rather than `n_retries` directly? Not sure if this is a stylistic/consistency decision. Feel free to reject the suggestion if it is.

```suggestion
    while 1:
        try:
            urlretrieve(remote.url, file_path)
            break
        except (URLError, TimeoutError):
            if n_retries > 0:
                warnings.warn(f"Retry downloading from url: {remote.url}")
                time.sleep(delay)
                n_retries -= 1
```